components/clp-mcp-server/clp_mcp_server/clp_mcp_server.py

```diff
@@ -34,10 +38,12 @@ def main(host: str, port: int, config_path: Path) -> int:
     :param config_path: The path to server's configuration file.
     :return: Exit code (0 for success, non-zero for failure).
     """
-    logging.basicConfig(
-        level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
-    )
-    logger = logging.getLogger(__name__)
+    # Setup logging to file
+    log_file_path = Path(os.getenv("CLP_LOGS_DIR")) / "mcp_server.log"
+    logging_file_handler = logging.FileHandler(filename=log_file_path, encoding="utf-8")
+    logging_file_handler.setFormatter(get_logging_formatter())
+    logger.addHandler(logging_file_handler)
+    set_logging_level(logger, os.getenv("CLP_LOGGING_LEVEL"))
```

Comment:
in a future PR, we should extract the `configure_logger` util from `garbage_collector` into `clp-py-utils` so we don't have to repeat these logger init steps in many different places in the pacakge:
https://github.com/y-scope/clp/blob/6abc934920b55da2de9c05ebf565cf96e47e6a45/components/job-orchestration/job_orchestration/garbage_collector/utils.py#L21-L28

@coderabbitai create an issue to track the refactor